### ISSUE TITLE
23694242: Provide a way to control the date format NSDatePicker uses

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#6867]
#### Description

Currently, NSDatePicker always uses the date style specified in the (system-wide) Language & Region system preferences pane. It is not easily possible to force a different format. My use case for this is to force a date picker to use 24-hour time in an app where that makes sense even though the vast majority of users will not want to use 24 hour time system-wide. It is possible to provide NSDatePicker with a formatter using the formatter property inherited from NSControl, however to my knowledge, this has no effect.

tl;dr: Please provide an API to force NSDatePicker to use a particular format to display its date/time.
## 

Product Version: Mac OS X 10.11.1 (15B42)
Created: 2015-11-30T21:55:45.163260
Originated: 2015-11-30T14:55:00
Open Radar Link: http://www.openradar.me/23694242
